### ISSUE TITLE
add option to report N valid alignments

### ISSUE THREAD

**Issue by pcantalupo** [boratyng/magicblast#4]
Hi,
I noticed that by default that MB reports more than 1 alignment for a query sequence as long as each alignment is valid. Is there an option to limit this to only 1 valid alignment? This would be similar to the default operation of Bowtie2 and the `-k <int>` option in [HISAT2](https://ccb.jhu.edu/software/hisat2/manual.shtml)

**Comment by boratyng** [comment]
Currently there is no such option. Sorry.  We are planning to add an option similar to ```-k <int>``` in Bowtie2 and Hisat2. 

Number of alignments is reported for each alignment in the NH tag in SAM output and in 18-th column of tabular output.

**Comment by boratyng** [comment]
Is this option needed for other reasons than speed or output file size? If there is more than one alignment with the same score Bowtie2 by default selects one at random.

**Comment by pcantalupo** [comment]
Yes, for the reasons you mentioned plus the hassle of having to manually remove them myself. Thank you
